ios/RNSScreenStack.mm

```diff
@@ -1178,6 +1174,25 @@ - (void)mountingTransactionWillMount:(react::MountingTransaction const &)transac
   }
 }
 
+- (void)mountingTransactionDidMount:(const facebook::react::MountingTransaction &)transaction
+               withSurfaceTelemetry:(const facebook::react::SurfaceTelemetry &)surfaceTelemetry
+{
+  for (auto &mutation : transaction.getMutations()) {
+    if (mutation.parentShadowView.componentName != nil &&
+        strcmp(mutation.parentShadowView.componentName, "RNSScreenStack") == 0) {
+      if (mutation.type == react::ShadowViewMutation::Type::Update ||
+          mutation.type == react::ShadowViewMutation::Type::Remove) {
```

Comment:
I would avoid nesting conditions if not necessary

Reply:
I did it intentionally, as four conditions in one if are kinda smudging in the eyes 😄

Reply:
For me it looks like separating such `if` means that there should be some `else` for the nested condition. Also, why do we call it on `update` and `remove` and not on `insert` and `remove` ?

Reply:
Because from my observation, `update` operations are being prepared after all `insert`/`remove` operations, which means that we should have ideal state for updating containers, since we're at the moment where all mutations have been performed and we're ready to update. However, I just noticed that it's only the case, when we're navigating to different screens, but not popping from some of them - I'll also check how this method behaves :)

Reply:
@WoLewicki Everything still works fine with `update` and `remove` operations, even if I'm doing some mayhem scenarios :smile:

Reply:
Original suggestion fixed in https://github.com/software-mansion/react-native-screens/pull/2249/commits/fb9e60ff78c9bb7f7f851f2ff9dea52e10dbc5d1.